lib/pbench/server/indexing_tarballs.py

```diff
@@ -0,0 +1,492 @@
+        except TemplateError as e:
+            idxctx.logger.error("update_templates [end], error {}", repr(e))
+            res = 9
+        except SigTermException:
+            # Re-raise a SIGTERM to avoid it being lumped in with general
+            # exception handling below.
+            raise
+        except Exception:
+            idxctx.logger.exception(
+                "update_templates [end]: Unexpected template" " processing error"
+            )
+            res = 12
+        else:
+            idxctx.logger.debug("update_templates [end]")
+            res = 0
+
+        if res != 0:
+            # Exit early if we encounter any errors.
+            return res
+
+        report = Report(
+            idxctx.config,
+            self.name,
+            es=idxctx.es,
+            pid=idxctx.getpid(),
+            group_id=idxctx.getgid(),
+            user_id=idxctx.getuid(),
+            hostname=idxctx.gethostname(),
+            version=VERSION,
+            templates=idxctx.templates,
+        )
+        # We use the "start" report ID as the tracking ID for all indexed
+        # documents.
+        try:
+            tracking_id = report.post_status(tstos(idxctx.time()), "start")
+        except SigTermException:
+            # Re-raise a SIGTERM to avoid it being lumped in with general
+            # exception handling below.
+            raise
+        except Exception:
+            idxctx.logger.error("Failed to post initial report status")
+            return 12
+        else:
+            idxctx.set_tracking_id(tracking_id)
+
+        with tempfile.TemporaryDirectory(
+            prefix=f"{self.name}.", dir=idxctx.config.TMP
+        ) as tmpdir:
+            idxctx.logger.debug("start processing list of tar balls")
+            tb_list = Path(tmpdir, f"{self.name}.{idxctx.TS}.list")
+            try:
+                with tb_list.open(mode="w") as lfp:
+                    # Write out all the tar balls we are processing so external
+                    # viewers can follow along from home.
+                    for size, controller, tb in tarballs:
+                        print(f"{size:20d} {controller} {tb}", file=lfp)
+
+                indexed = Path(tmpdir, f"{self.name}.{idxctx.TS}.indexed")
+                erred = Path(tmpdir, f"{self.name}.{idxctx.TS}.erred")
+                skipped = Path(tmpdir, f"{self.name}.{idxctx.TS}.skipped")
+                ie_filepath = Path(
+                    tmpdir, f"{self.name}.{idxctx.TS}.indexing-errors.json"
+                )
+
+                # We use a list object here so that when we close over this
+                # variable in the handler, the list object will be closed over,
+                # but not its contents.
+                sigquit_interrupt = [False]
+
+                def sigquit_handler(*args):
+                    sigquit_interrupt[0] = True
+
+                signal.signal(signal.SIGQUIT, sigquit_handler)
```

Comment:
Are we not disabling this signal handler anywhere? I don't see it. It probably doesn't really matter, as it only sets a flag and once we've seen the flag we'll never look again, but it would be cleaner (and certainly more modular) to disable the handler before we leave the `process_tb` scope.

Reply:
Good catch, seems like a gap left from the previous work.  It should be set to `SIG_IGN` by adding a `try/finally` block around the `for` loop starting on line 223.

Reply:
ok will update that